src/components/ValidateCode/ValidateCodeModal.tsx

```diff
@@ -10,29 +9,39 @@ import TextLink from '@components/TextLink';
 import useLocalize from '@hooks/useLocalize';
 import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import * as ValidationUtils from '@libs/ValidationUtils';
+import NotFoundPage from '@pages/ErrorPage/NotFoundPage';
 import variables from '@styles/variables';
 import * as Session from '@userActions/Session';
 import ONYXKEYS from '@src/ONYXKEYS';
-import type {Session as SessionType} from '@src/types/onyx';
 
-type ValidateCodeModalOnyxProps = {
-    /** Session of currently logged in user */
-    session: OnyxEntry<SessionType>;
-};
-
-type ValidateCodeModalProps = ValidateCodeModalOnyxProps & {
+type ValidateCodeModalProps = {
     /** Code to display. */
     code: string;
     /** The ID of the account to which the code belongs. */
     accountID: number;
 };
 
-function ValidateCodeModal({code, accountID, session = {}}: ValidateCodeModalProps) {
+function ValidateCodeModal({code, accountID}: ValidateCodeModalProps) {
     const theme = useTheme();
     const styles = useThemeStyles();
+    const [session] = useOnyx(ONYXKEYS.SESSION);
     const signInHere = useCallback(() => Session.signInWithValidateCode(accountID, code), [accountID, code]);
     const {translate} = useLocalize();
 
+    const isInvalidValidateCode = !ValidationUtils.isValidValidateCode(code);
+
+    if (isInvalidValidateCode) {
```

Comment:
@mkzie2 I prefer to use FullPageNotFoundView as a wrapper for cleaner. Could you help to update it?

Reply:
@DylanDylann I updated.